browser/js/common/factories/user.factory.js

```diff
@@ -0,0 +1,43 @@
+app.factory('UserFactory', function($http) {
+
+	var fetchAll = function() {
+		return $http.get('/api/users')
+			.then(res => res.data);
+	};
+	var fetchOne = function(userId) {
+		return $http.get('/api/users/' + userId)
+			.then(res => res.data);
+	};
+
+	var deleteOne = function(user) {
+		return $http.delete('/api/users/' + user._id)
+			.then(res => res.data);
+	};
+
+	var promoteToAdmin = function(user) {
+		user.admin = true;
+		return $http.put('/api/users/' + user._id, user)
+			.then(res => res.data);
+	};
+
+	var demoteToUser = function(user) {
+		user.admin = false;
+		return $http.put('/api/users/' + user._id, user)
```

Comment:
Don't you want to wait until the promise resolves before 'user.admin = false' on line 24?